apiary.apib

```diff
@@ -449,91 +868,370 @@ ipfs.io. TXT "dnslink=/ipfs/QmRzTuh2Lpuz7Gr39stNr6mTFdqAghsZec1JoUnfySUzcy ..."
 ## resolve [GET /name/resolve]
+Gets the value currently published at an IPNS name
+
++ Parameters
+    + arg (string, required) - The IPNS name to resolve. Defaults to your node's peerID.
+    + recursive (boolean, optional) - Resolve until the result is not an IPNS name
+    + nocache (boolean, optional) - Do not used cached entries
 
 + Response 200 (text/plain)
 
++ Response 500
+
+    + Attributes
+        + `Message`: "could not resolve name" (string)
+        + `Code`: 0 (number)
+
 # Group object
 Interact with ipfs objects.
 
-## data [GET /object/data]
+`ipfs object` is a plumbing command used to manipulate DAG objects
+directly.
 
-+ Response 200 (text/plain)
+## data [GET /object/data{?arg}]
+Outputs the raw bytes in an IPFS object
 
-## get [GET /object/get]
+`ipfs object data` is a plumbing command for retreiving the raw bytes stored in
+a DAG node. It outputs to stdout, and <key> is a base58 encoded
+multihash.
+
+Note that the "--encoding" option does not affect the output, since the
+output is the raw data of the object.
+
++ Parameters
+    + arg (string, optional) - Key of the object to retrieve, in base58-encoded multihash format
 
 + Response 200 (text/plain)
 
-## links [GET /object/links]
+## get [GET /object/get{?arg}{&encoding}]
+Get and serialize the DAG node named by <key>
+
+`ipfs object get` is a plumbing command for retreiving DAG nodes.
+It serializes the DAG node to the format specified by the "--encoding"
+flag. It outputs to stdout, and <key> is a base58 encoded multihash.
+
++ Parameters
+    + arg (string, required) - Key of the object to retrieve (in base58-encoded multihash format)
+    + encoding (enum[string], optional) - Serializes the DAG node to the format specified
+        + Default: `protobuf`
+        + Members
+            + `protobuf`
+            + `json`
+            + `xml`
 
 + Response 200 (text/plain)
 
-## new [GET /object/new]
+## links [GET /object/links{?arg}]
+Outputs the links pointed to by the specified object
+
+`ipfs object links` is a plumbing command for retreiving the links from
+a DAG node. It outputs to stdout, and <key> is a base58 encoded
+multihash.
+
++ Parameters
+    + arg (string, required) - Key of the object to retrieve, in base58-encoded multihash format
 
 + Response 200 (text/plain)
 
-## patch [GET /object/patch]
+## new [GET /object/new{?arg}]
+Creates a new object from an ipfs template
+
+`ipfs object new` is a plumbing command for creating new DAG nodes.
+By default it creates and returns a new empty merkledag node, but
+you may pass an optional template argument to create a preformatted
+node.
+
++ Parameters
+    + arg (string, required) - optional template to use
+
+        Available templates:
+            * unixfs-dir
+
 
 + Response 200 (text/plain)
 
-## put [GET /object/put]
+## patch [GET /object/patch{?arg1,arg2}{&name,ref}]
+Create a new merkledag object based on an existing one
+
+`ipfs object patch <root> <cmd> <args>` is a plumbing command used to
```

Comment:
Yeah, I don't know about this one. I wasn't able to get it to work on Postman. Ought we to remove it?